api/v1beta1/disruption_webhook_test.go

```diff
@@ -252,6 +252,74 @@ var _ = Describe("Disruption", func() {
 			})
 		})
 
+		Describe("spec.trigger possible errors", func() {
+			BeforeEach(func() {
+				k8sClient = makek8sClientWithDisruptionPod()
+				deleteOnly = false
+			})
+
+			JustBeforeEach(func() {
+				newDisruption = makeValidNetworkDisruption()
+				controllerutil.AddFinalizer(newDisruption, chaostypes.DisruptionFinalizer)
+			})
+
+			AfterEach(func() {
+				k8sClient = nil
+				newDisruption = nil
+			})
```

Comment:
Favour DeferCleanup usage to have cleanup close to setup.

I'm not sure why it's needed TBH as you are already re-creating those everytime.

Maybe, create both the client AND the disruption in the `BeforeEach` hence you will be able to modify them in other `BeforeEach` 

And you can remove the just before each (as finalizers can be moved to `BeforeEach` also IMO)